File: examples/canonical-auth/python/src/routes/permissions.py
from fastapi import APIRouter, Depends

from src.auth.claims import AuthContext
from src.auth.middleware import require_permission, store, get_auth_context

router = APIRouter()

@router.get("/permissions", dependencies=[Depends(require_permission("iam:permission:read"))])
def list_permissions(auth: AuthContext = Depends(get_auth_context)):
    return {"permissions": store.permissions}

File: examples/canonical-auth/typescript/src/routes/permissions.ts
import { Hono } from "hono";
import { jwtMiddleware, requirePermission } from "../auth/middleware.ts";
import { store } from "../domain/store.ts";

const app = new Hono();

app.use("/permissions", jwtMiddleware);

app.get("/permissions", requirePermission("iam:permission:read"), async (c) => {
  return c.json({ permissions: store.permissions });
});

export default app;
